crates/extensions/c8y_mapper_ext/src/log_upload.rs

```diff
@@ -0,0 +1,222 @@
+use crate::converter::CumulocityConverter;
+use crate::error::ConversionError;
+use crate::error::CumulocityMapperError;
+use crate::error::CumulocityMapperError::UnknownDevice;
+use c8y_api::smartrest::smartrest_deserializer::SmartRestLogRequest;
+use c8y_api::smartrest::smartrest_deserializer::SmartRestRequestGeneric;
+use c8y_api::smartrest::smartrest_serializer::CumulocitySupportedOperations;
+use c8y_api::smartrest::smartrest_serializer::SmartRestSerializer;
+use c8y_api::smartrest::smartrest_serializer::SmartRestSetOperationToExecuting;
+use c8y_api::smartrest::smartrest_serializer::SmartRestSetOperationToFailed;
+use c8y_api::smartrest::smartrest_serializer::SmartRestSetOperationToSuccessful;
+use c8y_api::smartrest::topic::C8yTopic;
+use nanoid::nanoid;
+use tedge_api::entity_store::EntityType;
+use tedge_api::messages::CommandStatus;
+use tedge_api::messages::LogMetadata;
+use tedge_api::messages::LogUploadCmdPayload;
+use tedge_api::mqtt_topics::Channel;
+use tedge_api::mqtt_topics::ChannelFilter::Command;
+use tedge_api::mqtt_topics::ChannelFilter::CommandMetadata;
+use tedge_api::mqtt_topics::EntityFilter::AnyEntity;
+use tedge_api::mqtt_topics::EntityTopicId;
+use tedge_api::mqtt_topics::MqttSchema;
+use tedge_api::mqtt_topics::OperationType;
+use tedge_api::Jsonify;
+use tedge_mqtt_ext::Message;
+use tedge_mqtt_ext::MqttMessage;
+use tedge_mqtt_ext::QoS;
+use tedge_mqtt_ext::TopicFilter;
+use tedge_utils::file::create_directory_with_defaults;
+use tedge_utils::file::create_file_with_defaults;
+
+pub fn log_upload_topic_filter(mqtt_schema: &MqttSchema) -> TopicFilter {
+    vec![
+        mqtt_schema.topics(AnyEntity, Command(OperationType::LogUpload)),
+        mqtt_schema.topics(AnyEntity, CommandMetadata(OperationType::LogUpload)),
+    ]
+    .into()
+}
+
+impl CumulocityConverter {
+    /// Convert a SmartREST logfile request to a Thin Edge log_upload command
+    pub fn convert_log_upload_request(
+        &self,
+        smartrest: &str,
+    ) -> Result<Vec<Message>, CumulocityMapperError> {
+        if !self.config.capabilities.log_management {
+            // Log_management is disabled
+            return Ok(vec![]);
+        }
+
+        let log_request = SmartRestLogRequest::from_smartrest(smartrest)?;
+        let target = self
+            .entity_store
+            .get_by_id(&log_request.device)
+            .ok_or_else(|| UnknownDevice {
+                device_id: log_request.device.to_string(),
+            })?;
+
+        let cmd_id = nanoid!();
+        let channel = Channel::Command {
+            operation: OperationType::LogUpload,
+            cmd_id: cmd_id.clone(),
+        };
+        let topic = self.mqtt_schema.topic_for(&target.topic_id, &channel);
+
+        let tedge_url = format!(
+            "http://{}/tedge/file-transfer/{}/log_upload/{}-{}",
+            &self.config.tedge_http_host, target.entity_id, log_request.log_type, cmd_id
```

Comment:
In the [spec](https://github.com/thin-edge/thin-edge.io/blob/main/docs/src/references/tedge-log-management.md#handling-log-upload-commands), I wrote to use **main** as the main device name, e.g. 
`http://127.0.0.1:8000/tedge/file-transfer/**main**/log_upload/mosquitto-1234`. However, I think it's better to use the entity ID of the main device as implemented here. This decision makes the implementation simpler.

So I will change the spec accordingly after this PR gets merged ;)

Reply:
PR is #2048